docs_src/microservices/application/AdvancedTopics.md

```diff
@@ -0,0 +1,426 @@
+# Advanced Topics
+
+The following items discuss topics that are a bit beyond the basic use cases of the Application Functions SDK when interacting with EdgeX.
+
+### Configurable Functions Pipeline
+
+This SDK provides the capability to define the functions pipeline via configuration rather than code by using the **app-service-configurable** application service. See the [App Service Configurable](./AppServiceConfigurable.md) section for more details.
```

Comment:
Great point - should definitely stay consistent. My inclination is keep it the same name of the repo, and thus the docker image in the compose which is "app-service-configurable".

Reply:
> @rsdmike - looks good with lots of detail as implied by 'advanced topics'. Mostly small spelling and grammar stuff.
> 
> I had a question on the below. Wasn't sure how not having a filter would cause a loop here. Not speaking to the correctness of the statement, but want to understand this potential issue. If it is true, would we not want to error the service out when it starts up?
> Be aware that without a filter in your pipeline, it is possible to create an infinite loop when the Message Bus trigger is used. Choose your device-name and reading name appropriately.

This can happen with anything that listens to core-data and then sends data back to core-data. Since any listener w/ ZMQ receives all data broadcasted by what it is subscribed to (ie. core-data), if you push something to core-data, it will publish on the bus again. If you don't have any filter -- that just keeps happening over and over again.

Reply:
OK - so you have to have the pipeline pushing it back into the system.  It is not just based on not filtering.  Got it.